File: src/main/java/com/example/demo/model/Beer.java
package com.example.demo.model;

import com.example.demo.utils.LocalDateDeserializer;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import lombok.*;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import java.net.URL;
import java.time.LocalDate;

import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;



@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@Entity
public class Beer {

    @Id
    private Long id;
    private String name;
    private String tagline;
    @Column(columnDefinition = "DATE")
    @JsonProperty("first_brewed")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "MM/yyyy")
    @JsonDeserialize(using = LocalDateDeserializer.class)
    @JsonSerialize(using = LocalDateSerializer.class)
    private LocalDate firstBrewed;
    @Column(columnDefinition = "TEXT")
    private String description;
    @JsonProperty("image_url")
    private String imageUrl;
    private Double abv;


    @Override
    public boolean equals(Object o){
        if(!(o instanceof Beer) ){
            return false;
        }
        Beer otherBeer = (Beer) o;
        return this.id.equals(otherBeer.getId());
    }
    @Override
    public int hashCode(){
        return id.hashCode();
    }

}

File: demo-ng/src/app/beer.ts
export class Beer {

    

    private _id!: number;
    private _name!: string;
    private _tagline!: string;
    private _first_brewed!: Date;
    private _description!: string;
    private _image_url!: string;
    private _abv!: number;

	constructor() {
	}

    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
    public get name(): string {
        return this._name;
    }
    public set name(value: string) {
        this._name = value;
    }
    public get tagline(): string {
        return this._tagline;
    }
    public set tagline(value: string) {
        this._tagline = value;
    }
    public get first_brewed(): Date {
        return this._first_brewed;
    }
    public set first_brewed(value: Date) {
        this._first_brewed = value;
    }
    public get description(): string {
        return this._description;
    }
    public set description(value: string) {
        this._description = value;
    }
    public get image_url(): string {
        return this._image_url;
    }
    public set image_url(value: string) {
        this._image_url = value;
    }
    public get abv(): number {
        return this._abv;
    }
    public set abv(value: number) {
        this._abv = value;
    }





    
}
